src/main.js

```diff
@@ -284,15 +304,10 @@ class MiniGraphCard extends LitElement {
           @click=${e => this.handlePopup(e, this.entity[id])}
           style=${entityConfig.state_adaptive_color ? `color: ${this.computeColor(state, id)};` : ''}>
           ${entityConfig.show_indicator ? this.renderIndicator(state, id) : ''}
-          <span class="state__value ellipsis">
-            ${this.computeState(isPrimary && tooltipValue || state)}
-          </span>
-          <span class="state__uom ellipsis">
-            ${this.computeUom(isPrimary && entity || id)}
-          </span>
+          ${this.unitFirstOrLast(isPrimary, tooltipValue, state, id, entity)}
```

Comment:
in case you like to have a look at nested templates, maybe something along these lines might work:
```js
unitFirstOrLast(state, unit) {
  if (this.config.unit_first) {
    return unit+state;
  } else {
    return state+unit;
  }
}
```

To be used as:
```
${ this.unitFirstOrLast(
  `state template`, 
  `unit template`
) }
```

This would avoid to repeat the span templates twice. But readability might suffer. I'm not sure.

Reply:
I'll take a proper look at this in the morning but I like the sound of that. Just for clarity: you mean to call `unitFirstOrLast` with the actual template `span` strings as the parameters? 

`unitFirstOrLast("<span>state</span>", "<span>unit</span>)`

I'm not actually sure that would impede readability; it'd keep the actual "style code" in the right place and just use the function to choose the order - so would probably improve it 😄

Reply:
Sorry, I just saw your comment in my mailbox. But, yes, that's what I meant!